src/Solver.h

```diff
@@ -126,9 +126,9 @@ class Solver {
   /** The total number of Tracks */
   int _tot_num_tracks;
 
-  /** An integer array with the Track uid separating the azimuthal and periodic
-   * halfspaces */
-  int* _num_tracks_by_halfspace;
+  /** An integer array with the Track uid separating the parallel track
```

Comment:
I found this comment confusing to read at first glance. Does the "uid" really add any relevant info here? I would remove that piece of the comment, or elaborate on it further if needed (if anything, one can't always assume that a uid is a monotonically increasing global integer counter as is the case for our Tracks).